internal/clients/telegram/state/handler.go

```diff
@@ -48,22 +57,30 @@ func NewTransition(
 	}
 }
 
-func Handle(bot update.User, upd update.Update, state Handler) Transition {
+func Handle(ctx context.Context, bot update.User, upd update.Update, state Handler) Transition {
 	if message, isSome := upd.Message.Unwrap(); isSome {
-		if transition, ok := handleMessage(bot, message, state); ok {
+		if transition, ok := handleMessage(ctx, bot, message, upd.ID, state); ok {
 			return transition
 		}
 	}
 
-	return state.Ignore()
+	if cq, isSome := upd.CallbackQuery.Unwrap(); isSome {
+		return state.CallbackQuery(ctx, cq)
+	}
+
+	logging.Infof("%s Ignoring this update using state.Ignore()", upd.ID.Log())
+
+	return state.Ignore(ctx)
 }
 
-func handleMessage(bot update.User, message update.Message, state Handler) (Transition, bool) {
+func handleMessage(ctx context.Context, bot update.User, message update.Message, updateID update.UpdateID,
+	state Handler,
+) (Transition, bool) {
 	switch message.Chat.Type {
 	case update.ChatTypePrivate:
 		text, isSome := message.Text.Unwrap()
 		if !isSome {
-			return Transition{}, false
+			return Transition{}, false //nolint:exhaustruct // False indicates an error
```

Comment:
just turn off this linter for full project `exhaustruct`

Reply:
There are a couple of structs that should be linted but I guess its ok to just turn it off completely.

Done.

Reply:
I dont know if Go has a way to exclude issues by source. They only have it for linter error texts.

Reply:
https://golangci-lint.run/usage/false-positives/